### DISCUSSION TITLE
Can login to my Github but im not authorized to make any push commits

### DISCUSSION THREAD

**Discussion by NathanielKinderman** [community/community#22446] [New to GitHub]
<p>Hello,<br>
So I’m having trouble making any push commits to any repo. When trying to push my commit, I get the login prompt and fail every time. I can login in my GitHub profile with same credentials just fine. I’ve changed my password, tried the Oauth tokens and stack overflow.<br>
Originally, I was working with on a team and was trying to commit to my coworkers repo and failed. Then, I created a test repo and try to push a simple HTML file to the repo and failed. So I don’t what the next step is, any help would be appreciated.<br>
Thank you.</p>

**Comment by byrneh** [comment, marked as answer]
<p>If you are trying to use the same password as used to login to your user profile (web browser GUI), you need to change Authentication method to web auth flow, personal access token or SSH Keys for working with GIi.</p>
<p>Use a <strong>recent/latest version</strong> of Git Client or GitHub Desktop version you can simply select the ‘Sign In with Browser option’ and follow the instructions.</p>
<p>Alternatively the personal access token routes requires you to create it in advance, described at this link <a href="https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc">Creating a personal access token - GitHub Docs</a>. Examples of how you enter the token in the Git command line are provided in the link also.</p>

**Reply by rhavin** [reply]
"you need to change Authentication method to web auth flow, personal access token or SSH Keys for working with GIi."

..where do I do this?

**Comment by NathanielKinderman** [comment]
<p>Thank you for the help! It worked!</p>

**Comment by byrneh** [comment]
<p>That is great news, if you mark the above post as ‘Solution’, that will help others viewing this post</p>

**Comment by philtymps** [comment]
<p>I’m using Eclipse and can’t figure out how to make the push/commit work.  it keeps popping up a a login dialog and i’ve tried providing user/password and user/personal access token but neither works.  After entering the login details I get not authorized error.</p>

**Reply by pooja19purohit** [reply]
Install EGit on Eclipse and set git username and email

**Comment by kingthorin** [comment]
<p><a class="mention" href="/philtymps">@philtymps</a> make sure you’re trying to push to your repo (fork).</p>

**Comment by TDesjardins** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc" title="08:28PM - 15 December 2020">The GitHub Blog – 15 Dec 20</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/1841306/181070957-0c319b03-3f78-4eda-8557-4bf414421534.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11f8abd6e6e16e4af4babafe405da97b4856127a_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11f8abd6e6e16e4af4babafe405da97b4856127a_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/1/11f8abd6e6e16e4af4babafe405da97b4856127a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11f8abd6e6e16e4af4babafe405da97b4856127a_2_10x10.png"></div>

<h3><a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a></h3>

  <p>Beginning August 13th, 2021, we will no longer accept account passwords when authenticating Git operations on GitHub.com.</p>

  <p>
    <span class="label1">Est. reading time: 3 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by LineBreaker30113** [comment]
I am using Eclipse and I set my user name and password accordingly but it gives me not authorized error. Please tell me either how to make the git use my github password or please tell how to learn my other password.

**Reply by kingthorin** [reply]
In Eclipse you'd use username and a Personal Access Token (PAT).

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
